packages/phoenix-ng/projects/phoenix-app/cypress/e2e/cms.test.ts

```diff
@@ -0,0 +1,171 @@
+describe('CMS', () => {
+  it('should match the whole page', () => {
+    cy.visit('/cms').then(() => {
+      cy.contains('This may take a while').should('not.be.visible');
+      // in order to make the detector appear which loads in 2 - 2.5 seconds
+      cy.wait(2500);
+      cy.document().toMatchImageSnapshot();
+    });
+  });
+
+  describe('Phoenix Menu', () => {
+    it('should have options to Save and Load state', () => {
+      cy.get('.phoenix-menu-item-right .item-settings').click();
+
+      cy.contains('Save state').should('exist');
+      cy.contains('Load state').should('exist');
+
+      cy.get('.phoenix-menu-item-right .item-config-backdrop').click();
+    });
+
+    it('should display correct options under event data dropdown', () => {
+      cy.get('.phoenix-menu-item-right > button').eq(0).click();
+      cy.get('.phoenix-menu-item-right > button').eq(1).click();
+
+      cy.contains('Tracks').should('exist');
+      cy.contains('Jets').should('exist');
+      cy.contains('CaloClusters').should('exist');
+      cy.contains('MuonChambers').should('exist');
+
+      cy.get('.phoenix-menu-item-right > button').eq(1).click();
+    });
+
+    it('should hide event data', () => {
+      cy.get('app-phoenix-menu-item .mat-slide-toggle').eq(0).click();
+
+      cy.document().toMatchImageSnapshot();
+
+      cy.get('app-phoenix-menu-item').eq(1).click();
+    });
+
+    it('should hide detector geometry', () => {
+      cy.get('app-phoenix-menu-item .mat-slide-toggle').eq(2).click();
+
+      cy.document().toMatchImageSnapshot();
+
+      cy.get('app-phoenix-menu-item').eq(3).click();
+      cy.get('.phoenix-menu-item-right > button').eq(0).click();
+    });
+  });
+
+  describe('Phoenix Iconbar', () => {
+    it('should switch to use dark theme', () => {
+      cy.get('app-dark-theme').eq(0).click();
+
+      cy.document().toMatchImageSnapshot();
+
+      cy.get('app-dark-theme').eq(0).click();
+    });
+
+    it('should zoom out', () => {
+      cy.get('app-zoom-controls app-menu-toggle > button').eq(0).focus();
+      for (let n = 0; n < 20; ++n) {
+        cy.focused().click();
+      }
+      cy.document().toMatchImageSnapshot();
+    });
+
+    it('should do object clipping properly', () => {
+      cy.get('app-object-clipping').click();
+      cy.contains('Clipping').click();
```

Comment:
**Try to scope all the selectors that use `contains` or if possible, avoid using `contains` as a selector.** It seems very prone to breaking tests later on.